Intermittent failing test: BrowserThumbnailsTest.clearThumbnails

We're seeming a test failure (`BrowserThumbnailsTest.clearThumbnails`) that passed and was merged by CI. I can get it to fail consistently but it must have been an intermittent to have slipped passed CI:

```

java.lang.IllegalStateException: cache is closed

	at com.jakewharton.disklrucache.DiskLruCache.checkNotClosed(DiskLruCache.java:614)
	at com.jakewharton.disklrucache.DiskLruCache.get(DiskLruCache.java:405)
	at mozilla.components.browser.thumbnails.utils.ThumbnailDiskCache.getThumbnailData$browser_thumbnails_debug(ThumbnailDiskCache.kt:41)
	at mozilla.components.browser.thumbnails.storage.ThumbnailStorage.loadThumbnailInternal(ThumbnailStorage.kt:88)
	at mozilla.components.browser.thumbnails.storage.ThumbnailStorage.access$loadThumbnailInternal(ThumbnailStorage.kt:34)
	at mozilla.components.browser.thumbnails.storage.ThumbnailStorage$loadThumbnail$1.invokeSuspend(ThumbnailStorage.kt:68)
	at |b|b|b(Coroutine boundary.|b(|b)
	at kotlinx.coroutines.DeferredCoroutine.await$suspendImpl(Builders.common.kt:99)
	at mozilla.components.browser.thumbnails.storage.ThumbnailStorageTest$clearThumbnails$1.invokeSuspend(ThumbnailStorageTest.kt:46)
Caused by: java.lang.IllegalStateException: cache is closed
	at com.jakewharton.disklrucache.DiskLruCache.checkNotClosed(DiskLruCache.java:614)
	at com.jakewharton.disklrucache.DiskLruCache.get(DiskLruCache.java:405)
	at mozilla.components.browser.thumbnails.utils.ThumbnailDiskCache.getThumbnailData$browser_thumbnails_debug(ThumbnailDiskCache.kt:41)
	at mozilla.components.browser.thumbnails.storage.ThumbnailStorage.loadThumbnailInternal(ThumbnailStorage.kt:88)
	at mozilla.components.browser.thumbnails.storage.ThumbnailStorage.access$loadThumbnailInternal(ThumbnailStorage.kt:34)
	at mozilla.components.browser.thumbnails.storage.ThumbnailStorage$loadThumbnail$1.invokeSuspend(ThumbnailStorage.kt:68)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)


```

--- a/components/browser/thumbnails/src/test/java/mozilla/components/browser/thumbnails/storage/ThumbnailStorageTest.kt
+++ b/components/browser/thumbnails/src/test/java/mozilla/components/browser/thumbnails/storage/ThumbnailStorageTest.kt
@@ -11,6 +11,7 @@ import kotlinx.coroutines.runBlocking
 import mozilla.components.support.test.ext.joinBlocking
 import mozilla.components.support.test.mock
 import mozilla.components.support.test.robolectric.testContext
+import mozilla.components.support.test.rule.createTestCoroutinesDispatcher
 import org.junit.After
 import org.junit.Assert.assertEquals
 import org.junit.Assert.assertNotNull
@@ -24,6 +25,8 @@ import org.mockito.Mockito.spy
 @RunWith(AndroidJUnit4::class)
 class ThumbnailStorageTest {
 
+    val testDispatcher = createTestCoroutinesDispatcher()
+
     @Before
     @After
     fun cleanUp() {
@@ -33,7 +36,7 @@ class ThumbnailStorageTest {
     @Test
     fun `clearThumbnails`() = runBlocking {
         val bitmap: Bitmap = mock()
-        val thumbnailStorage = spy(ThumbnailStorage(testContext))
+        val thumbnailStorage = spy(ThumbnailStorage(testContext, testDispatcher))
 
         thumbnailStorage.saveThumbnail("test-tab1", bitmap).joinBlocking()
         thumbnailStorage.saveThumbnail("test-tab2", bitmap).joinBlocking()
@@ -53,7 +56,7 @@ class ThumbnailStorageTest {
     fun `deleteThumbnail`() = runBlocking {
         val sessionIdOrUrl = "test-tab1"
         val bitmap: Bitmap = mock()
-        val thumbnailStorage = spy(ThumbnailStorage(testContext))
+        val thumbnailStorage = spy(ThumbnailStorage(testContext, testDispatcher))
 
         thumbnailStorage.saveThumbnail(sessionIdOrUrl, bitmap).joinBlocking()
         var thumbnail = thumbnailStorage.loadThumbnail(sessionIdOrUrl).await()
@@ -68,7 +71,7 @@ class ThumbnailStorageTest {
     fun `saveThumbnail`() = runBlocking {
         val sessionIdOrUrl = "test-tab1"
         val bitmap: Bitmap = mock()
-        val thumbnailStorage = spy(ThumbnailStorage(testContext))
+        val thumbnailStorage = spy(ThumbnailStorage(testContext, testDispatcher))
         var thumbnail = thumbnailStorage.loadThumbnail(sessionIdOrUrl).await()
 
         assertNull(thumbnail)
@@ -82,7 +85,7 @@ class ThumbnailStorageTest {
     fun `loadThumbnail`() = runBlocking {
         val sessionIdOrUrl = "test-tab1"
         val bitmap: Bitmap = mock()
-        val thumbnailStorage = spy(ThumbnailStorage(testContext))
+        val thumbnailStorage = spy(ThumbnailStorage(testContext, testDispatcher))
 
         thumbnailStorage.saveThumbnail(sessionIdOrUrl, bitmap)
         `when`(thumbnailStorage.loadThumbnail(sessionIdOrUrl)).thenReturn(CompletableDeferred(bitmap))

